src/object/srv_obj.c

```diff
@@ -2844,6 +2844,17 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 			       &orw->orw_oid, NULL, 0, dtx_flags, NULL, &dth);
 		if (rc == 0) {
 			rc = obj_local_rw(rpc, &ioc, dth);
+			if (rc != 0)
+				D_CDEBUG(
+					rc == -DER_INPROGRESS || rc == -DER_TX_RESTART ||
```

Comment:
we really need a better solution for these. I have similar code in a macro in VOS (VOS_TX*).   But every one of these means multiple checks